src/common/admin_socket.cc

```diff
@@ -189,8 +189,14 @@ std::string AdminSocket::bind_and_listen(const std::string &sock_path, int *fd)
   // FIPS zeroization audit 20191115: this memset is fine.
   memset(&address, 0, sizeof(struct sockaddr_un));
   address.sun_family = AF_UNIX;
+#ifdef __linux__
+  address.sun_path[0] = '\0';
+  snprintf(address.sun_path+1, sizeof(address.sun_path),
```

Comment:
this'd be a security issue. i'd suggest root cause the problem instead.

Reply:
I don't think it's a security issue/problem, beginning the name with a NUL is how you use the abstract socket namespace.

Reply:
quote from unix(7)

 

> Abstract sockets
>        Socket permissions have no meaning for abstract sockets: the process
>        umask(2) has no effect when binding an abstract socket, and changing
>        the ownership and permissions of the object (via fchown(2) and
>        fchmod(2)) has no effect on the accessibility of the socket.
> 
>        Abstract sockets automatically disappear when all open references to
>        the socket are closed.
> 
>        The abstract socket namespace is a nonportable Linux extension.

Reply:
How do abstract sockets interact with kernel namespaces? Does each filesystem namespace get its own?

If not, then it seems like you could have conflicts when two daemons are running on the box in different containers.